testing/adios2/engine/bp/operations/TestBPWriteReadZfp.cpp

```diff
@@ -411,12 +415,11 @@ void ZfpRate3D(const double rate)
     }
 }
 
-void ZfpRate1DSel(const double rate)
+void ZFPRate1DSel(const std::string rate)
```

Comment:
Maybe this is bikeshedding, but shouldn't this be a reference?

Reply:
Using C++11 Small String Optimization https://stackoverflow.com/questions/10315041/meaning-of-acronym-sso-in-the-context-of-stdstring/10319672#10319672

Reply:
Hmm, it doesn't look to me that there is any optimization being performed:

https://godbolt.org/z/edaqCc

The generated asm is identical either way. In any case, the rationale that it documents an expectation of small strings is very reasonable.

Reply:
That's not how SSO works. In the example above the compiler is never aware of string size. it just functions passing by value or reference. SSO means that if the size of the string is known the compiler will [put it in the stack](https://shaharmike.com/cpp/std-string/) passing it by value is the same reason we pass primitives by value.